src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,137 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @Test
+    void registerUserNull_NotOk() {
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(null);
+        }, "An exception should be thrown");
+
+        User bobAgeNull = new User();
+        bobAgeNull.setAge(null);
+        bobAgeNull.setLogin("bobist");
+        bobAgeNull.setPassword("hhjfbchbv");
+        bobAgeNull.setId(52698L);
+
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(bobAgeNull);
+        }, "An exception should be thrown");
+
+        User bobLoginNull = new User();
+        bobLoginNull.setAge(19);
+        bobLoginNull.setLogin(null);
+        bobLoginNull.setPassword("hhjfbchbv");
+        bobLoginNull.setId(52698L);
+
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(bobLoginNull);
+        }, "An exception should be thrown");
+
+        User bobPasswordNull = new User();
+        bobPasswordNull.setAge(19);
+        bobPasswordNull.setLogin("bobist");
+        bobPasswordNull.setPassword("");
+        bobPasswordNull.setId(52698L);
+
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(bobPasswordNull);
+        }, "An exception should be thrown");
+

```

Comment:
You test have naming nullUser, but I see 3 kinds of user and no one user is null!!! divide to few tests